### PR TITLE
Update Onboarding

### LINKED ISSUE #182: Remove onboarding and user distinction
**Description:** Nowadays, we offer 3 onboardings (company, community and professional). These lead to unnecessary complexity in early days of product. They should be inexistent and only an onboarding iwith Entity name and Fiscal number should  exists needed right now. All users should be the same and have same functionalities. 

**Acceptance Criteria:** Onboard removed from front end and backend. Keep only two fields

**Requirements:** None

**Context:** The user should have no distinction. All are the same.

**User Stories (if applicable add images):** See image

### LINKED ISSUE #189: Store onboarding data and show warning if professional doesn’t exist.
- [x] Save name, generic tax information, and address.
- [x] Checkbox: I agree with terms and Conditions.
- [x] When creating a contract, display a warning if the professional's past address is not registered

### PR DESCRIPTION
**Pull request by nomadbitcoin**
- **Unregistered Professional Address Warning (feat)**: A new feature alerts users if the professional address they enter is not registered, enhancing the UX by providing immediate feedback without blocking form submission.

- **Onboarding and User Selection Removal (chore)**: Removed outdated onboarding and user selection code to declutter the codebase.

- **Generic Onboarding Component (feat)**: Implemented a new generic onboarding component that efficiently captures and stores basic user data within the contracts, simplifying the user's initial setup experience.

These updates collectively refine the user journey, reduce unnecessary complexity.

Closes #189 
Closes #182

### CODE REVIEW

**Comment by danicuki** [comment]
Screen Shot 2023-11-28 at 13 21 24
what is this huge checkbox???

**Comment by danicuki** [comment]
about the storage of tax document, we should not store the plain data, but a hashed version of it (with proper salta)

**Comment by nomadbitcoin** [comment]
> 

there is a bug with all the checkbox styles in the application, I have already reported it to Mika and he will apply the default style, more info on #194

File: components/AddAgreement/AddAgreement.js [hunk 1]
```diff
@@ -124,6 +153,13 @@ function AddAgreementForm(props) {
           return validator && validator.test(value)
         }
       )
+      .test(
+        "check-professional-registered",
+        async function (value) {
+          checkUser(value);
+          return true; // Retorne true para que o teste não falhe independentemente do resultado
```
Comment (danicuki) [inline components/AddAgreement/AddAgreement.js:160]: comment in English

File: pages/_app.js [hunk 2]
```diff
@@ -20,14 +23,79 @@ function PageContent({ Component, pageProps }) {
...
+      throw error;
+    }
+  };
+
+  useEffect(() => {
+    if (!account || !contract) {
+      return;
+    }
+  
+    if (router.pathname.startsWith("/onboarding")) {
```
Comment (danicuki) [inline pages/_app.js:86]: this if could be inside the previous one

**Review by danicuki — APPROVED** [review]

**Pull request merged**